Document/0x05e-Testing-Cryptography.md

```diff
@@ -291,6 +291,74 @@ When defining the KeyDescription AuthorizationList, the following parameters wil
+            "3082028b30820232a003020102020900a2059ed10e435b57300a06082a8648ce3d040302308198310b30090603550406130255533113301106035504080c0a43616c69666f726e69613116301406035504070c0d4d6f756e7461696e205669657731153013060355040a0c0c476f6f676c652c20496e632e3110300e060355040b0c07416e64726f69643133303106035504030c2a416e64726f6964204b657973746f726520536f667477617265204174746573746174696f6e20526f6f74301e170d3136303131313030343335305a170d3336303130363030343335305a308198310b30090603550406130255533113301106035504080c0a43616c69666f726e69613116301406035504070c0d4d6f756e7461696e205669657731153013060355040a0c0c476f6f676c652c20496e632e3110300e060355040b0c07416e64726f69643133303106035504030c2a416e64726f6964204b657973746f726520536f667477617265204174746573746174696f6e20526f6f743059301306072a8648ce3d020106082a8648ce3d03010703420004ee5d5ec7e1c0db6d03a67ee6b61bec4d6a5d6a682e0fff7f490e7d771f44226dbdb1affa16cbc7adc577d2569caab7b02d54015d3e432b2a8ed74eec487541a4a3633061301d0603551d0e04160414c8ade9774c45c3a3cf0d1610e479433a215a30cf301f0603551d23041830168014c8ade9774c45c3a3cf0d1610e479433a215a30cf300f0603551d130101ff040530030101ff300e0603551d0f0101ff040403020284300a06082a8648ce3d040302034700304402203521a3ef8b34461e9cd560f31d5889206adca36541f60d9ece8a198c6648607b02204d0bf351d9307c7d5bda35341da8471b63a585653cad4f24a7e74daf417df1bf"
+        ]
+    }
+}
+```
+
+In the above JSON snippet, the keys have the following meaning:
+        _fmt_: Attestation statement format identifier
+        _authData_: It denotes the authenticator data for the attestation
+        _alg_: The algorithm that is used for the Signature
+        _sig_: Signature
+        _x5c_: Attestation certificate chain
+
+Note: The _sig_ is generated by concatenating _authData_ and _clientDataHash_(challenge sent by the server) and signing through the credential private key using the alg signing algorithm and the same is verified at the server-side by using the public key in the first certificate
+
+For more understanding on the implementation guidelines, [Google Sample Code]("https://github.com/googlesamples/android-key-attestation/blob/master/server/src/main/java/com/android/example/KeyAttestationExample.java" "Google Sample Code For Android Key Attestation") can be referred.
+
+For the security analysis perspective the analysts may perform the following checks for the secure implementation of Key Attestation:
+
+* Check if the key attestation is totally implemented at the client-side. In such scenario, the same can be easily bypassed by tampering the application, method hooking, etc.
+
+* Check if the server uses challenge while initiating the key attestation. As failing to do that would lead to insecure implementation thus making it vulnerable to replay attacks. Also, checks pertaining to the randomness of the challenge should be performed.
```

Comment:
Check if the server uses challenge --> uses a random challenge with a Nonce

Reply:
Why do we need a nonce with a challenge? I think I didn't get what you are trying to convey.

Reply:
Please ignore, it is not in their instructions either and the protocol seems a bit different.